src/librbd/io/SimpleSchedulerObjectDispatch.cc

```diff
@@ -105,7 +105,8 @@ bool SimpleSchedulerObjectDispatch<I>::ObjectRequests::try_delay_request(
 
     // try to merge back to an existing request
     iter = m_delayed_requests.lower_bound(object_off);
-    if (iter == m_delayed_requests.end() || iter->first > object_off) {
+    if (iter != m_delayed_requests.begin() &&
+        (iter == m_delayed_requests.end() || iter->first > object_off)) {
```

Comment:
Good catch!  Please spawn this into a separate PR with a tracker ticket for backporting.

Reply:
Sure, right away.